Kitodo-API/src/main/java/org/kitodo/config/Config.java

```diff
@@ -41,20 +42,26 @@ static PropertiesConfiguration getConfig(String configFile) {
             synchronized (Config.class) {
                 if (!configMap.containsKey(configFile)) {
                     PropertiesConfiguration initialized;
-                    AbstractConfiguration.setDefaultListDelimiter('&');
                     try {
-                        initialized = new PropertiesConfiguration(configFile);
+                        // Create and initialize the builder
+                        ReloadingFileBasedConfigurationBuilder<PropertiesConfiguration> builder =
+                            new ReloadingFileBasedConfigurationBuilder<>(PropertiesConfiguration.class)
+                                .configure(new Parameters().properties()
+                                    .setFileName(configFile)
+                                    .setThrowExceptionOnMissing(true)
+                                    .setListDelimiterHandler(new DefaultListDelimiterHandler('&'))
+                                    .setIncludesAllowed(false));
+                        // Register an event listener for triggering reloading checks
+                        builder.addEventListener(ConfigurationBuilderEvent.CONFIGURATION_REQUEST,
+                            (EventListener) event -> builder.getReloadingController().checkForReloading(null));
```

Comment:
```suggestion
                            event -> builder.getReloadingController().checkForReloading(null));
```
Is this cast necessary? I first extended it to `(EventListener<ConfigurationBuilderEvent>) event` because the IDE marked it as an "Unchecked assignment" but then it would tell me that the cast is redundant altogether. I removed the typecast and it still works.

Reply:
The `EventListener` casting is not needed. Thanks for hinting this.